Settings.c

```diff
@@ -73,20 +73,27 @@ static void writeQuotedList(FILE* fp, char** list) {
 
 */
 
-void Settings_delete(Settings* this) {
-   free(this->filename);
-   free(this->initialFilename);
-   for (unsigned int i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
+static void Settings_deleteColumns(Settings* this) {
+   for (size_t i = 0; i < HeaderLayout_getColumns(this->hLayout); i++) {
       String_freeArray(this->hColumns[i].names);
       free(this->hColumns[i].modes);
    }
    free(this->hColumns);
```

Comment:
Should check for `this->hColumns` being non-NULL. For consistency with `Settings_deleteScreens`.

Reply:
Again, I'm not making functional changes here.  IIRC there is a legit case where NULL screens is used so I'm keeping the code as-is and not changing any logic like this.